src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -926,3 +928,194 @@ def acquireTiledArea(streams, stage, area, overlap=0.2, settings_obs=None, log_p
+                self._future.running_subf = autofocus_in_roi(roi,
+                                                             self._stage,
+                                                             self._ccd,
+                                                             self._focus,
+                                                             self.focus_rng,
+                                                             self.n_focus_points,
+                                                             self.conf_level)
+
+                logging.debug(f"Autofocus is running for roi number {idx} with {roi} values")
+
+                try:
+                    focus_points = self._future.running_subf.result()  # timeout error can be used
+                except Exception as exp:
+                    # logging.exception(
+                    #     f"Autofocus within roi failed for roi number {idx} with {roi} values due to {exp}")
+                    raise
+
+                # cancel the sub future
+                if self._future._task_state == CANCELLED:
+                    raise CancelledError()
+
+                # run tiled acquisition for the selected roi
+                self._future.running_subf = acquireTiledArea(self.streams, self._stage, roi,
+                                                             focus_range=self.focus_rng,
+                                                             overlap=self._overlap,
+                                                             settings_obs=self._settings_obs,
+                                                             log_path=self._log_path,
+                                                             zlevels=self._zlevels,
+                                                             registrar=self._registrar,
+                                                             weaver=self._weaver,
+                                                             focusing_method=self.focusing_method,
+                                                             focus_points=focus_points)
+                logging.debug(f"Z-stack acquisition is running for roi number {idx} with {roi} values")
+                try:
+                    da = self._future.running_subf.result()
+                    # append dataArrays
+                    da_rois.append(da[0])
+                except Exception as exp:
+                    logging.exception(
+                        f"Z-stack acquisition within roi failed for roi number {idx} with {roi} values due to {exp}")
+                    raise
+
+                logging.debug(f"acquisition overview is completed for roi number {idx} with {roi} values")
+
+                # cancel the sub future
+                if self._future._task_state == CANCELLED:
+                    raise CancelledError()
+
+                # Store the actual time during acquisition one roi
+                actual_time_per_roi = time.time() - start_time
```

Comment:
I don't think there will be a lot of variations, but instead of using just the duration of the latest RoI acquired, you could do the average: move `start_time` to before the for loop, and then compute as:
```suggestion
                actual_time_per_roi = (time.time() - start_time) / (idx + 1)
```